### PR TITLE
fix: Merklerootmnlist calc bls scheme

### PR DESCRIPTION
**Pull request by ogabrielides**
## Issue being fixed or feature implemented

 
Code was still using legacy scheme for `pubKeyOperator` of each `CSimplifiedMNListEntry` while calculating `MerklerootMNList` while v19 HF was active.

## What was done?

When building `CSimplifiedMNList` for `MerklerootMNList`, this PR adds the correct set of `nVersion` for `CSimplifiedMNListEntry` so that basic scheme will be used instead of legacy.

Furthermore, DIP3 unit tests suites for v19 will test for 1000 blocks instead of 900 to cover v19 activated period.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by UdjinM6 — DISMISSED** [review]
utACK

File: src/evo/simplifiedmns.cpp [hunk 1]
```diff
@@ -92,13 +93,17 @@ CSimplifiedMNList::CSimplifiedMNList(const std::vector<CSimplifiedMNListEntry>&
     });
 }
 
-CSimplifiedMNList::CSimplifiedMNList(const CDeterministicMNList& dmnList)
+CSimplifiedMNList::CSimplifiedMNList(const CDeterministicMNList& dmnList, const CBlockIndex* pindexPrev)
 {
     mnList.resize(dmnList.GetAllMNsCount());
 
+    bool v19active = llmq::utils::IsV19Active(pindexPrev);
```
Comment (PastaPastaPasta) [inline src/evo/simplifiedmns.cpp:100]: I don't think this is a good spot to be doing the check.. Why don't we pass in a boolean here representing whether v19 is active? that way we don't have to include llmq/utils here as well.

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: src/evo/simplifiedmns.cpp (hunk 1 above)
Comment (ogabrielides) [reply src/evo/simplifiedmns.cpp]: Sure

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

**Pull request merged**